src/commands/importFunctionApp/importFunctionApp.ts

```diff
@@ -0,0 +1,319 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ApiContract, BackendContract, BackendCredentialsContract, OperationCollection, OperationContract, PropertyContract } from "azure-arm-apimanagement/lib/models";
+import { WebSiteManagementClient } from "azure-arm-website";
+import { Site, WebAppCollection } from "azure-arm-website/lib/models";
+import { ProgressLocation, window } from "vscode";
+import xml = require("xml");
+import { FunctionAppService } from "../../Azure/WebApp/FunctionAppService";
+import { IFunctionContract } from "../../Azure/WebApp/IFunctionContract";
+import * as Constants from "../../constants";
+import { ApisTreeItem } from "../../explorer/ApisTreeItem";
+import { ApiTreeItem } from "../../explorer/ApiTreeItem";
+import { ServiceTreeItem } from "../../explorer/ServiceTreeItem";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../localize";
+import { apiUtil } from "../../utils/apiUtil";
+import { nonNullOrEmptyValue } from "../../utils/nonNull";
+import { parseUrlTemplate } from "./parseUrlTemplate";
+
+export async function importFunctionAppToApi(node?: ApiTreeItem): Promise<void> {
+    if (!node) {
+        node = <ApiTreeItem>await ext.tree.showTreeItemPicker(ApiTreeItem.contextValue);
+    }
+
+    ext.outputChannel.show();
+    ext.outputChannel.appendLine(localize("importFunctionApp", `Import function app to api ${node.root.apiName} started...`));
+
+    const funcAppService = new FunctionAppService();
+    const client = funcAppService.getClient(node.root.credentials, node.root.subscriptionId, node.root.environment);
+    const allfunctionApps = await listFunctionApps(client);
+    const functionApp = await pickFunctionApp(allfunctionApps);
+    const funcName = nonNullOrEmptyValue(functionApp.name);
+    const funcAppResourceGroup = nonNullOrEmptyValue(functionApp.resourceGroup);
+    const functionAppBaseUrl = funcAppService.getSiteUrl(node.root.environment.resourceManagerEndpointUrl, node.root.subscriptionId, funcAppResourceGroup, funcName);
```

Comment:
This can be internal to the FunctionAppService class. And FunctionAppService class can take node.root.environment.resourceManagerEndpointUrl, node.root.subscriptionId, funcAppResourceGroup, funcName as part of the constructor